src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -19,6 +19,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Inherits AbstractSyntaxFactsService
         Implements ISyntaxFactsService
 
+        Public Shared ReadOnly Property [Default] As New VisualBasicSyntaxFactsService
```

Comment:
Ok, when I implement your suggestion like so.
```vb.net
    <ExportLanguageServiceFactory(GetType(ILanguageServiceFactory), LanguageNames.VisualBasic), [Shared]>
    Public Class VisualBasicSyntaxFactsServiceFactory
        Implements ILanguageServiceFactory
        Public Sub New()

        End Sub

        Public Function CreateLanguageService(languageServices As HostLanguageServices) As ILanguageService Implements ILanguageServiceFactory.CreateLanguageService
            Return New VisualBasicSyntaxFactsService(languageServices)
        End Function

        <ExportLanguageService(GetType(ISyntaxFactsService), LanguageNames.VisualBasic), [Shared]>
        Friend Class VisualBasicSyntaxFactsService
            Inherits AbstractSyntaxFactsService
            Implements ISyntaxFactsService

            Friend Sub New(LanguageService As HostLanguageServices)
                Me.LanguageServices = LanguageServices
            End Sub

            Private ReadOnly Property LanguageServices As HostLanguageServices
```
Usage Example
```vb.net
        <Import(NameOf(VisualBasicSyntaxFactsServiceFactory.VisualBasicSyntaxFactsService))>
        Private Property _Facts As VisualBasicSyntaxFactsServiceFactory.VisualBasicSyntaxFactsService


        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsSuppression)>
        Public Async Function TestSuppressionWithAttributeOnEveryNodes() As Threading.Tasks.Task
            Await TestSuppressionWithAttributeAsync(
                TestResource.AllInOneVisualBasicCode,
                VisualBasicParseOptions.Default,
                digInto:=Function(n)
                             Dim member = _Facts.GetContainingMemberDeclaration(n, n.Span.Start)
                             If member Is Nothing OrElse member Is n Then
                                 Return True
                             End If

                             Return Not TypeOf n Is StatementSyntax
                         End Function,
                verifier:=Function(t) t.IndexOf("SuppressMessage", StringComparison.Ordinal) >= 0)
        End Function
```

I get thousand and thousands of errors. Of the two following categories of type of errors.
```
Collection: Test collection for ConvertToAutoPropertyVB.UnitTests.ConvertToAutoPropertyTests

Test: ConvertToAutoPropertyVB.UnitTests.ConvertToAutoPropertyTests.TestCommaSeparatedField3
Exception Type: System.Composition.Hosting.CompositionFailedException
Message:
System.Composition.Hosting.CompositionFailedException : No importing constructor was found on type 'Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxFactsServiceFactory+VisualBasicSyntaxFactsService'.
   at System.Composition.TypedParts.Discovery.DiscoveredPart.<GetPartActivatorDependencies>d__d.MoveNext()
   at System.Linq.Enumerable.<ConcatIterator>d__58`1.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at System.Composition.TypedParts.Discovery.DiscoveredPart.GetDependencies(DependencyAccessor definitionAccessor)
   at System.Composition.TypedParts.Discovery.DiscoveredExport.<>c__DisplayClass2.<GetExportDescriptorPromise>b__0()
   at System.Composition.Hosting.Core.ExportDescriptorPromise.<>c__DisplayClass4.<.ctor>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.CheckTarget(CompositionDependency dependency, HashSet`1 checked, Stack`1 checking)
   at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.CheckDependency(CompositionDependency dependency, HashSet`1 checked, Stack`1 checking)
   at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.CheckTarget(CompositionDependency dependency, HashSet`1 checked, Stack`1 checking)
   at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.CheckDependency(CompositionDependency dependency, HashSet`1 checked, Stack`1 checking)
   at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.CheckTarget(CompositionDependency dependency, HashSet`1 checked, Stack`1 checking)
   at System.Composition.Hosting.Core.ExportDescriptorRegistryUpdate.Execute(CompositionContract contract)
   at System.Composition.Hosting.Core.ExportDescriptorRegistry.TryGetSingleForExport(CompositionContract exportKey, ExportDescriptor& defaultForExport)
   at System.Composition.Hosting.Core.LifetimeContext.TryGetExport(CompositionContract contract, Object& export)
   at System.Composition.Hosting.CompositionHost.TryGetExport(CompositionContract contract, Object& export)
   at System.Composition.CompositionContextExtensions.SatisfyImportsInternal(CompositionContext exportProvider, Object objectWithLooseImports, AttributedModelProvider conventions)
   at System.Composition.CompositionContextExtensions.SatisfyImports(CompositionContext compositionContext, Object objectWithLooseImports)
   at Microsoft.CodeAnalysis.Host.Mef.MefHostServices.Microsoft.CodeAnalysis.Host.Mef.IMefHostExportProvider.GetExports[TExtension,TMetadata]() in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Workspace\Host\Mef\MefHostServices.cs:line 61
   at Microsoft.CodeAnalysis.Host.Mef.MefLanguageServices..ctor(MefWorkspaceServices workspaceServices, String language) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Workspace\Host\Mef\MefLanguageServices.cs:line 28
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.<>c__DisplayClass21_0.<GetLanguageServices>b__0(String _) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Workspace\Host\Mef\MefWorkspaceServices.cs:line 160
   at System.Collections.Immutable.ImmutableInterlocked.GetOrAdd[TKey,TValue](ImmutableDictionary`2& location, TKey key, Func`2 valueFactory)
   at Microsoft.CodeAnalysis.Host.Mef.MefWorkspaceServices.GetLanguageServices(String languageName) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Workspace\Host\Mef\MefWorkspaceServices.cs:line 160
   at Microsoft.CodeAnalysis.Solution.AddProject(ProjectInfo projectInfo) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Workspace\Solution\Solution.cs:line 598
   at Microsoft.CodeAnalysis.Solution.AddProject(ProjectId projectId, String name, String assemblyName, String language) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Workspace\Solution\Solution.cs:line 569
   at Roslyn.UnitTestFramework.CodeActionProviderTestFixture.CreateDocument(String code) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Samples\Shared\UnitTestFramework\CodeActionProviderTestFixture.cs:line 30
   at Roslyn.UnitTestFramework.CodeRefactoringProviderTestFixture.Test(String markup, String expected, Int32 actionIndex, Boolean compareTokens) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Samples\Shared\UnitTestFramework\CodeRefactoringProviderTestFixture.cs:line 62
   at ConvertToAutoPropertyVB.UnitTests.ConvertToAutoPropertyTests.TestCommaSeparatedField3() in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Samples\VisualBasic\ConvertToAutoProperty\Test\ConvertToAutoPropertyTests.vb:line 114
```
```
Test: Microsoft.CodeAnalysis.Editor.UnitTests.Rename.RenameEngineTests.RenameMainWithoutAssertFailureVB
Exception Type: System.AggregateException
Message:
System.AggregateException : One or more errors occurred.
---- System.NullReferenceException : Object reference not set to an instance of an object.
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result()
   at Microsoft.CodeAnalysis.Editor.UnitTests.Rename.RenameEngineResult.Create(ITestOutputHelper helper, XElement workspaceXml, String renameTo, Dictionary`2 changedOptionSet) in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\EditorFeatures\Test2\Rename\RenameEngineResult.vb:line 101
   at Microsoft.CodeAnalysis.Editor.UnitTests.Rename.RenameEngineTests.RenameMainWithoutAssertFailureVB() in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\EditorFeatures\Test2\Rename\RenameEngineTests.vb:line 5746
----- Inner Stack Trace -----
   at Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.<FindSymbolAtPositionAsync>d__2.MoveNext() in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\FindSymbols\SymbolFinder.cs:line 55
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.FindSymbols.SymbolFinder.<FindSymbolAtPositionAsync>d__3.MoveNext() in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\FindSymbols\SymbolFinder.cs:line 74
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.Rename.RenameLocations.ReferenceProcessing.<GetRenamableSymbolAsync>d__0.MoveNext() in C:\Users\Adam\Source\Repos\roslyn-AdamSpeight2008\src\Workspaces\Core\Portable\Rename\RenameLocation.ReferenceProcessing.cs:line 33
```

I can understand the second category arising, if the mef import isn't being satisfied.